llvm/lib/Target/X86/X86InstrInfo.cpp

```diff
@@ -3443,6 +3443,8 @@ static bool isX87Reg(unsigned Reg) {
 
 /// check if the instruction is X87 instruction
 bool X86::isX87Instruction(MachineInstr &MI) {
+  if (MI.isCall())
+    return false;
```

Comment:
If you'd like to insert wait after call instruction, code should be added in X86InsertWait.cp, instead of `X86::isX87Instruction`.

Reply:
Could you add comment to explain this, e.g. ".Call defs X87 register but should not be classified as X87 instruction."?

Reply:
> If you'd like to insert wait after call instruction, code should be added in X86InsertWait.cp, instead of `X86::isX87Instruction`.

We add it before a call but not after.

We expect the function being called to have a wait before returning.